### PR TITLE
[CDAP-20690] Add unit tests for diff algorithm

### PR DESCRIPTION
**Pull request by mertkaya1033**
# [CDAP-20690](https://cdap.atlassian.net/browse/CDAP-20690) Add unit tests for diff algorithm

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [x] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-20690](https://cdap.atlassian.net/browse/CDAP-20690)

## Screenshots

[CDAP-20690]: https://cdap.atlassian.net/browse/CDAP-20690?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CDAP-20690]: https://cdap.atlassian.net/browse/CDAP-20690?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/cdap/components/PipelineDiff/util/__tests__/diff.tests.ts [hunk 1]
```diff
@@ -0,0 +1,298 @@
+import { DiffIndicator } from 'components/PipelineDiff/types';
```
Comment (sumengwang) [inline app/cdap/components/PipelineDiff/util/__tests__/diff.tests.ts:17]: copyright header

File: app/cdap/components/PipelineDiff/util/__tests__/pipelines.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+const getStageDiffKeyFromStageName = (name) => {
+  return {
+    [exampleStage1.name]:exampleStage1DiffKey,
+    [exampleStage2.name]:exampleStage2DiffKey,
+    [exampleStage3.name]:exampleStage3DiffKey
+  }[name]
+} 
+
+export const fromExample1ToExample2DiffKey = getConnectionDiffKey(fromExample1ToExample2, getStageDiffKeyFromStageName)
+export const fromExample1ToExample3DiffKey = getConnectionDiffKey(fromExample1ToExample3, getStageDiffKeyFromStageName)
```
Comment (sumengwang) [inline app/cdap/components/PipelineDiff/util/__tests__/pipelines.js]: newline

File: app/cdap/components/PipelineDiff/util/__tests__/diff.tests.ts [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+            diff: exampleStage3,
+            stage1: exampleStage3,
+            stage2: undefined,
+            diffIndicator: DiffIndicator.DELETED,
+          },
+        },
+      });
+    });
+
+    it('should indicate modification of a stage when a stage from pipeline1 is modified to get pipeline2', () => {
```
Comment (sumengwang) [inline app/cdap/components/PipelineDiff/util/__tests__/diff.tests.ts:168]: what's pipeline3, your description did not cover it

File: app/cdap/components/PipelineDiff/util/__tests__/diff.tests.ts (hunk 1 above)
Comment (mertkaya1033) [reply app/cdap/components/PipelineDiff/util/__tests__/diff.tests.ts:17]: done

File: app/cdap/components/PipelineDiff/util/__tests__/diff.tests.ts (hunk 3 above)
Comment (mertkaya1033) [reply app/cdap/components/PipelineDiff/util/__tests__/diff.tests.ts:168]: `pipeline1` and `pipeline2` refers to the arguments of `computePipelineDiff` function. I updated the `describe` to indicate this and changed all constant names to `examplePipelineX`.

File: app/cdap/components/PipelineDiff/util/__tests__/pipelines.js (hunk 2 above)
Comment (mertkaya1033) [reply app/cdap/components/PipelineDiff/util/__tests__/pipelines.js]: done

**Review by de-lan — APPROVED** [review]

**Pull request merged**